### PR TITLE
Avoid search for hypotheses and use refine instead

### PR DESCRIPTION
**Pull request by pi8027**

### CODE REVIEW

File: theories/lra.elpi [hunk 1]
```diff
@@ -100,8 +100,8 @@ rfstr.prop {{ @eq lp:Bool lp:Ty1 lp:Ty2 }} C IsField :-
 rfstr.prop {{ @eq lp:Ty _ _ }} C IsField :- !, type->carrier Ty C IsField.
 
 pred rfstr.hyps i:list prop, o:carrier, o:prop.
-rfstr.hyps [decl _ _ H|_] C IsField :- rfstr.prop H C IsField.
 rfstr.hyps [_|Ctx] C IsField :- rfstr.hyps Ctx C IsField.
+rfstr.hyps [decl _ _ H|_] C IsField :- rfstr.prop H C IsField.
```
Comment (gares) [inline theories/lra.elpi:104]: why don't you rev the list instead?

Comment (pi8027) [reply theories/lra.elpi:104]: I rather removed `std.rev` (in L.456). Do you think it is better to use rev?

Comment (pi8027) [reply theories/lra.elpi:104]: Since I did not see why we need to take the reversal of `Ctx` for `quote.goal`, I wanted to remove it entirely.

Comment (gares) [reply theories/lra.elpi:104]: I never tested the perf of exists+rev v.s. the code you wrote which is more heavy on backtracking....
I just find it surprising

Comment (pi8027) [reply theories/lra.elpi:104]: Sure. I will fix.

**Pull request merged**